woocommerce/woocommerce-gateway-stripe
# Currency-restricted UPE method admin notices

# Changes proposed in this Pull Request:

- [x] Add unit tests for existing admin notice logic.
- [x] Add support for unsupported currency notice when using UPE methods that require a specific currency not being used by the store.
- [x] Add unit tests for newly added UPE-related logic.

![](https://caps.evan.pro/498fc6.png)

# Testing instructions

- Enable UPE feature flag.
- Set store currency to USD (or anything besides EUR).
- Enable UPE in stripe settings.
- Enable UPE methods like Giropay, EPS, Bancontact, or iDeal.
- There should be notice(s) as shown in the above screenshot.
- Clicking the X on the notices should hide them.
